astropy/coordinates/builtin_frames/galactocentric.py

```diff
@@ -41,13 +43,21 @@ class galactocentric_frame_defaults(ScienceState):
     ``Galactocentric`` or ``Galactocentric()`` with no explicit arguments.
 
     This class controls the parameter settings by specifying a string name,
-    which can be one of:
+    which has the pre-specified options:
 
     - 'pre-v4.0': The current default value, which sets the default frame
       attribute values to their original (pre-astropy-v4.0) values.
     - 'v4.0': The attribute values as updated in Astropy version 4.0.
     - 'latest': An alias of the most recent parameter set (currently: 'v4.0')
 
+    Alternatively, user-defined parameter settings may be registered and set.
+
+    This class also tracks the references for all parameter values in the
+    attribute ``_references``, as well as any further information in
```

Comment:
With your new `references` class property, we can change it here. I think for `_state` we should either also make it public (i.e., have a `state` class property) or not mention it; I'd lean towards making it public as I quite like the idea.

Reply:
My two hesitations with making `_state` public are:

1. that the subsequent ScienceState PR will make each field of `_state` accessible as an attribute.
As a side note, this will actually obviate the need for the "value" and "references" class properties, though I think it would be more explicit to keep them around, regardless of strict necessity).

2. `_state` is mutable. This could be resolved using `MappingProxyType`, but I'll need to check whether there are downstream consequences. For instance, I'm looking at `Galactocentric` which appears to expect mutability of "references".

Reply:
Good arguments both. Since you went with my other suggestion of returning only `state` in the explicit `get_solar_params_from_string` method, we also do not really need it any more. So, maybe just refer to that as a way to get the state?

Reply:
Given the API change to returning the whole state, what about deprecating the method name `get_solar_params_from_string` in favor of `get_from_registry`?

```
@deprecated("v4.2", message="renamed `get_from_registry`")
@classmethod
def get_solar_params_from_string(cls, arg):
    """..."""
    return cls.get_from_registry(arg)
```

Reply:
That seems sensible to me!

Reply:
Yes, makes sense, though in that case I would let `get_solar_params_from_string` return what it originally did, so that working code doesn't break but can be adjusted.

Reply:
Done, `get_solar_params_from_string` calls `get_from_registry` and returns the `parameters` from `_state`